fluentd/datadog_checks/fluentd/fluentd.py

```diff
@@ -81,9 +88,34 @@ def check(self, instance):
                     # Filter unspecified plugins to keep backward compatibility.
                     if len(plugin_ids) == 0 or p.get('plugin_id') in plugin_ids:
                         self.gauge('fluentd.%s' % (m), metric, [tag] + custom_tags)
+
+            self._collect_metadata()
         except Exception as e:
             msg = "No stats could be retrieved from %s : %s" % (url, str(e))
             self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL, tags=service_check_tags, message=msg)
             raise
         else:
             self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.OK, tags=service_check_tags)
+
+    def _collect_metadata(self):
+        raw_version = self._get_raw_version()
+
+        if raw_version:
+            self.set_metadata('version', raw_version)
+
+    def _get_raw_version(self):
+        command = self._fluentd_command + ['--version']
+
+        try:
+            out, _, _ = get_subprocess_output(command, self.log, False)
```

Comment:
It seems the the api doesn't provide currently the version, but can be added easily upstream here, example: 

```
      def config_json(req)
        obj = {
          'pid' => Process.pid,
          'ppid' => Process.ppid
          'version' => Fluent::VERSION
        }.merge(@agent.fluentd_opts)
        opts = build_option(req)

        render_json(obj, pretty_json: opts[:pretty_json])
      end
```

https://github.com/fluent/fluentd/blob/master/lib/fluent/plugin/in_monitor_agent.rb#L65-L68

Even if we push this change upstream, it will be available only for new versions, so we will still need to retrieve versions via command line for old versions.

Reply:
Yup. I'll open an issue (and forward with a PR) on the fluentd repo to discuss adding the version to their monitoring REST API.

Agreed that this is not a blocker w.r.t. this PR, though?

Reply:
@AlexandreYang I opened https://github.com/fluent/fluentd/issues/2705 and https://github.com/fluent/fluentd/pull/2706 on the Fluentd repo.

Will discuss there whether supporting accessing a remote Fluentd instance makes any sense at all from a monitoring perspective. I'm now less sure that it can actually be a possible setup: most examples from the [Fluentd monitoring docs](https://docs.fluentd.org/deployment/monitoring) show calls to `localhost` or similar. If people always only install the integration on the host where Fluentd is installed, then the benefits we'd get from using the API are less clear IMO.